lib/diagnostic.js

```diff
@@ -29,8 +34,15 @@ const sum = function (/* previous, current */) {
 // in the array returns a truthy value
 // you may use any feature of JavaScript
 // other than `Array.prototype.some`
-const any = function (/* array, callback */) {
-
+const any = function (array, callback) {
+  for (let i = 0; i < array.length; i++) {
+    if (callback(array[i], i, array)) {
+      return true;
+    }
+    else {
+      return undefined;
+    }
+  }
 };
```

Comment:
I think what you were trying to write is this:
```
const any = function (array, callback) {
  for (let i = 0; i < array.length; i++) {
    if (callback(array[i])) {
      return true;
    }
  }
  return false
};
```

Here's another possible solution:
```
const any = function (array, callback) {
  return !array.every((e, i, a) => !callback(e, i, a));
};
```